File: AI-Local-Growth-SaaS/backend/app/schemas/ai.py
"""AI 接口 Schema。"""
from __future__ import annotations

from pydantic import BaseModel, Field


class CommentIn(BaseModel):
    video: str = Field(..., min_length=1, description="视频链接或描述文本")
    industry: str = Field(default="", description="行业")


class CommentOut(BaseModel):
    comments: list[str] = Field(default_factory=list)
    task_id: int


class ReportIn(BaseModel):
    merchant_id: int = Field(..., description="商家 id")
    store_id: int | None = Field(default=None, description="门店 id（可选）")


class ReportItem(BaseModel):
    dimension: str
    finding: str
    suggestion: str


class ReportOut(BaseModel):
    report: "ReportData"
    task_id: int


class ReportData(BaseModel):
    score: int
    summary: str
    items: list[ReportItem] = Field(default_factory=list)


class ContentIn(BaseModel):
    type: str = Field(default="script", description="script/copy")
    industry: str = Field(default="", description="行业")
    topic: str = Field(..., min_length=1, description="主题")
    tone: str | None = Field(default=None, description="语气风格（可选）")


class ContentOut(BaseModel):
    content: str
    task_id: int


ReportOut.model_rebuild()

File: AI-Local-Growth-SaaS/frontend/src/api/ai.ts
import http from "./http";

export interface CommentResult {
  comments: string[];
  task_id: number;
}

export interface ReportItem {
  dimension: string;
  finding: string;
  suggestion: string;
}

export interface ReportData {
  score: number;
  summary: string;
  items: ReportItem[];
}

export interface ReportResult {
  report: ReportData;
  task_id: number;
}

export interface ContentResult {
  content: string;
  task_id: number;
}

export interface CommentIn {
  video: string;
  industry?: string;
}

export interface ReportIn {
  merchant_id: number;
  store_id?: number | null;
}

export interface ContentIn {
  type: "script" | "copy";
  industry?: string;
  topic: string;
  tone?: string | null;
}

export function generateComment(body: CommentIn): Promise<CommentResult> {
  return http.post("/ai/comment", body) as Promise<CommentResult>;
}

export function generateReport(body: ReportIn): Promise<ReportResult> {
  return http.post("/ai/report", body) as Promise<ReportResult>;
}

export function generateContent(body: ContentIn): Promise<ContentResult> {
  return http.post("/ai/content", body) as Promise<ContentResult>;
}
